compute/compute-node.Dockerfile

```diff
@@ -1041,6 +1041,24 @@ RUN wget https://github.com/pgpartman/pg_partman/archive/refs/tags/v5.1.0.tar.gz
     make -j $(getconf _NPROCESSORS_ONLN) install && \
     echo 'trusted = true' >> /usr/local/pgsql/share/extension/pg_partman.control
 
+#########################################################################################
+#
+# Layer "pg_mooncake"
+# compile pg_mooncake extension
+#
+#########################################################################################
+FROM rust-extensions-build AS pg-mooncake-build
+ARG PG_VERSION
+COPY --from=pg-build /usr/local/pgsql/ /usr/local/pgsql/
+
+ENV PG_MOONCAKE_TAG=neon
```

Comment:
I wonder, should we refer to a specific commit? The branch name looks scary from the security PoV

Reply:
It does, but they don't have any tags last I checked. I could reference a single commit, but it would be a little more work.

Reply:
> I could reference a single commit, but it would be a little more work.

Why? We do this for the `pg_session_jwt` https://github.com/neondatabase/neon/blob/9909551f47e86d39e2df1cdeba79774e87744f17/compute/compute-node.Dockerfile#L978

You can download an archive as per any specific commit, just link is different

Reply:
No because there are submodules that need cloning

Reply:
I see, then something like that should work
```
git clone --branch neon --depth 1 https://github.com/Mooncake-Labs/pg_mooncake.git pg_mooncake-src
cd pg_mooncake-src
git fetch d030e60c480aee569e147075d885e3242b50eae1
git checkout d030e60c480aee569e147075d885e3242b50eae1
git submodule update --init --force --recursive
```

Could be likely shortened nad/or get use of better flags

Reply:
Ah, I see you did it in a similar way already

Reply:
I should add --recursive for good measure. I'll add that when I get to my laptop.